### PR TITLE
Suggest and auto fill title and description based on image location

### LINKED ISSUE #259: When user takes a photo, send push notification - "Is this a picture of A/B/C? Is it in administrative area D/E/F? (then link Wikidata/Commons)"
How about when the user take an image to upload, the location is checked against Wikidata items without image nearby (say, 200m), and then presents a list of all these under the heading "Is this a picture of..."?
Likewise, items near the location can give you a "administrative area", or at least a very short list thereof; follow that Wikidata list to "higher" units, and present the lowest one with a Commons category, as a "location category".

### PR DESCRIPTION
**Pull request by maskaravivek**
**Description (required)**

Fixes #259

What changes did you make and why?

**Tests performed (required)**

Tested on prod Debug

**Screenshots showing what changed (optional - for UI changes)**

Screenshot_2020-01-12-20-18-21-860_fr free nrw commons

### CODE REVIEW

**Comment by maskaravivek** [comment]
@misaochan If everything works perfectly for you, I will go ahead and add test cases. :)

**Comment by neslihanturan** [comment]
Hey @maskaravivek , I couldn't trigger this. Can you email me the example picture you used?

**Comment by maskaravivek** [comment]
@neslihanturan It should trigger for any picture with geolocation if that picture has a nearby item that needs a picture within a radius of 0.1 kilometers. For testing, you can go to `UploadRemoteDataSource` and change `NEARBY_RADIUS_IN_KILO_METERS` to a larger value like 2-5 km so that it triggers for any image that you have. :)

Also, make sure that your image has EXIF info and it is not getting stripped off because of app settings.

**Comment by neslihanturan** [comment]
Hey @maskaravivek , I was testing this PR. The app failed when I choose the image from gallery. Does this seemed related to you? :
```
020-01-27 13:22:00.800 30778-30778/fr.free.nrw.commons E/AndroidRuntime: FATAL EXCEPTION: main
    Process: fr.free.nrw.commons, PID: 30778
    java.lang.ClassCastException: java.util.HashSet cannot be cast to java.lang.String
        at android.app.SharedPreferencesImpl.getString(SharedPreferencesImpl.java:255)
        at fr.free.nrw.commons.kvstore.BasicKvStore.getString(BasicKvStore.java:96)
        at fr.free.nrw.commons.kvstore.BasicKvStore.getString(BasicKvStore.java:76)
        at fr.free.nrw.commons.kvstore.JsonKvStore.getJson(JsonKvStore.java:61)
        at fr.free.nrw.commons.upload.FileProcessor.getExifTagsToRedact(FileProcessor.java:108)
        at fr.free.nrw.commons.upload.FileProcessor.processFileCoordinates(FileProcessor.java:84)
        at fr.free.nrw.commons.upload.UploadModel.getUploadItem(UploadModel.java:142)
        at fr.free.nrw.commons.upload.UploadModel.preProcessImage(UploadModel.java:119)
        at fr.free.nrw.commons.repository.UploadRemoteDataSource.preProcessImage(UploadRemoteDataSource.java:175)
        at fr.free.nrw.commons.repository.UploadRepository.preProcessImage(UploadRepository.java:183)
        at fr.free.nrw.commons.upload.mediaDetails.UploadMediaPresenter.receiveImage(UploadMediaPresenter.java:78)
        at fr.free.nrw.commons.upload.mediaDetails.UploadMediaDetailFragment.init(UploadMediaDetailFragment.java:155)
        at fr.free.nrw.commons.upload.mediaDetails.UploadMediaDetailFragment.onViewCreated(UploadMediaDetailFragment.java:131)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:896)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:1229)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:1295)
        at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:686)
        at androidx.fragment.app.FragmentManagerImpl.executeOps(FragmentManagerImpl.java:2057)
        at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:1847)
        at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:1802)
        at androidx.fragment.app.FragmentManagerImpl.execSingleAction(FragmentManagerImpl.java:1679)
        at androidx.fragment.app.BackStackRecord.commitNowAllowingStateLoss(BackStackRecord.java:536)
        at androidx.fragment.app.FragmentStatePagerAdapter.finishUpdate(FragmentStatePagerAdapter.java:170)
        at androidx.viewpager.widget.ViewPager.populate(ViewPager.java:1244)
        at androidx.viewpager.widget.ViewPager.populate(ViewPager.java:1092)
        at androidx.viewpager.widget.ViewPager.onMeasure(ViewPager.java:1622)
        at android.view.View.measure(View.java:22002)
        at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:715)
        at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:461)
        at android.view.View.measure(View.java:22002)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6580)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at androidx.appcompat.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:143)
        at android.view.View.measure(View.java:22002)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6580)
        at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1514)
        at android.widget.LinearLayout.measureVertical(LinearLayout.java:806)
        at android.widget.LinearLayout.onMeasure(LinearLayout.java:685)
        at android.view.View.measure(View.java:22002)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6580)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at android.view.View.measure(View.java:22002)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6580)
        at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1514)
        at android.widget.LinearLayout.measureVertical(LinearLayout.java:806)
        at android.widget.LinearLayout.onMeasure(LinearLayout.java:685)
        at android.view.View.measure(View.java:22002)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6580)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
2020-01-27 13:22:00.800 30778-30778/fr.free.nrw.commons E/AndroidRuntime:     at com.android.internal.policy.DecorView.onMeasure(DecorView.java:721)
        at android.view.View.measure(View.java:22002)
        at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:2410)
        at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1498)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1751)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1386)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6733)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:911)
        at android.view.Choreographer.doCallbacks(Choreographer.java:723)
        at android.view.Choreographer.doFrame(Choreographer.java:658)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
        at android.os.Handler.handleCallback(Handler.java:789)
        at android.os.Handler.dispatchMessage(Handler.java:98)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
2020-01-27 13:22:00.806 30778-30778/fr.free.nrw.commons E/ACRA: ACRA caught a ClassCastException for fr.free.nrw.commons
    java.lang.ClassCastException: java.util.HashSet cannot be cast to java.lang.String
        at android.app.SharedPreferencesImpl.getString(SharedPreferencesImpl.java:255)
        at fr.free.nrw.commons.kvstore.BasicKvStore.getString(BasicKvStore.java:96)
        at fr.free.nrw.commons.kvstore.BasicKvStore.getString(BasicKvStore.java:76)
        at fr.free.nrw.commons.kvstore.JsonKvStore.getJson(JsonKvStore.java:61)
        at fr.free.nrw.commons.upload.FileProcessor.getExifTagsToRedact(FileProcessor.java:108)
        at fr.free.nrw.commons.upload.FileProcessor.processFileCoordinates(FileProcessor.java:84)
        at fr.free.nrw.commons.upload.UploadModel.getUploadItem(UploadModel.java:142)
        at fr.free.nrw.commons.upload.UploadModel.preProcessImage(UploadModel.java:119)
        at fr.free.nrw.commons.repository.UploadRemoteDataSource.preProcessImage(UploadRemoteDataSource.java:175)
        at fr.free.nrw.commons.repository.UploadRepository.preProcessImage(UploadRepository.java:183)
        at fr.free.nrw.commons.upload.mediaDetails.UploadMediaPresenter.receiveImage(UploadMediaPresenter.java:78)
        at fr.free.nrw.commons.upload.mediaDetails.UploadMediaDetailFragment.init(UploadMediaDetailFragment.java:155)
        at fr.free.nrw.commons.upload.mediaDetails.UploadMediaDetailFragment.onViewCreated(UploadMediaDetailFragment.java:131)
```

**Comment by maskaravivek** [comment]
@neslihanturan I am not getting this error.

@ashishkumar468 Can this be related to your recent changes in #3339?

**Comment by ashishkumar468** [comment]
@maskaravivek  Yes it is, @neslihanturan might have installed your build over one of my builds, anyways my changes are not merged on master, instead they are on 2.12, so should not affect your PR

@neslihanturan  Try a fresh install on this branch or if we want to test how the app would behave on update, do it over master or from the latest commit from 2.12, I guess your existing apk would be from one of my old commits

File: app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java [hunk 1]
```diff
@@ -297,6 +299,32 @@ public void onImageProcessed(UploadItem uploadItem, Place place) {
...
+     */
+    @SuppressLint("StringFormatInvalid")
+    @Override
+    public void onNearbyPlaceFound(UploadItem uploadItem, Place place) {
+        DialogUtil.showAlertDialog(getActivity(),
+                getString(R.string.upload_nearby_place_found_title),
+                String.format(Locale.getDefault(),
+                        getString(R.string.upload_nearby_place_found_description),
+                        place.getName()),
+                () -> {
```
Comment (ashishkumar468) [inline app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java]: Is this intended to be empty?

Comment (maskaravivek) [reply app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java]: yup. there's no action on cancel. the dialog is dismissed automatically.

**Comment by neslihanturan** [comment]
Fresh installation did the trick, it works for me. Thanks a lot @maskaravivek :)

**Pull request merged**